DS-Assignment6/UniqueCharacter.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Assignment 2
+ * Class to calculate the number of unique characters in a string
+ * Using hash map to cache the string previously encountered
+ * @author Arpit Pittie
+ */
+package session6;
+
+import java.util.HashMap;
+import java.util.Scanner;
+import java.util.TreeSet;
+
+public class UniqueCharacter {
+
+	private HashMap<String, Integer> characterUnique;	//Map to cache the result
+	private TreeSet<Character> characters;	//Set to store the unique elements in a string
+	private Scanner scan;	//to take input from the user
+	
+	public UniqueCharacter() {
+		characterUnique = new HashMap<String, Integer>();
+		scan = new Scanner(System.in);
+	}
+	
+	/**
+	 * Method to count the number of unique characters in a string
+	 * @param value - The string in which the unique character number is to be found
+	 * @return The number of unique characters in the string
+	 */
+	private int countUniqueCharacter(String value) {

```

Comment:
As this method is called internally by the other methods. Hence the members outside the class could not see its functionality.